src/cache_cf.cc

```diff
@@ -2390,8 +2390,10 @@ parse_peer(CachePeer ** head)
         p->connect_fail_limit = 10;
 
 #if USE_CACHE_DIGESTS
-    if (!p->options.no_digest)
-        peerDigestCreate(p);
+    if (!p->options.no_digest) {
+        auto pd = new PeerDigest(p);
+        p->digest = pd;
+    }
```

Comment:
I agree, but that look is a _natural_ (perhaps even welcomed!) outcome of a reference loop IMO. FWIW, I consider the earlier two-line version similarly "recursive", with the additional "why did they name that temporary pointer?" question/red flag.

AFAICT, the only way to avoid or clarify that "look" is to add a separate "linking" function that would set both pointers in both classes, but that would leave PeerDigest without a peer for a while, and that would be worse than the current code IMO.